src/aegis_ai/tools/http_get/__init__.py

```diff
@@ -0,0 +1,116 @@
+
+import requests
+
+from typing import Optional, Literal
+from pydantic import BaseModel, Field
+from pydantic_ai import Tool, RunContext
+
+from aegis_ai.tools import default_tool_http_headers
+
+logger = logging.getLogger(__name__)
+
+# set allow list that http get will allow retrieval from
+allow_list_string: str = os.getenv("AEGIS_HTTP_GET_TOOL_ALLOW_LIST", "")
+allow_list = []
+for host in allow_list_string.split(","):
+    if host.strip():
+        allow_list.append(host.strip().lower())
+
+
+class GetURLContentInput(BaseModel):
+    """Input schema for the get_url_content tool."""
+
+    url: str = Field(
+        ...,
+        description="The full URL (including 'http://' or 'https://') to retrieve content from.",
+    )
+
+
+class URLContent(BaseModel):
+    """
+    Structured context retrieved from a given URL using HTTP GET request.
+    """
+
+    requested_url: str = Field(..., description="The URL that was requested.")
+    status_code: Optional[int] = Field(
+        None, description="The HTTP status code of the response (e.g., 200, 404, 500)."
+    )
+    content: Optional[str] = Field(
+        None,
+        description="The content (body) of the HTTP response, typically HTML, JSON, or plain text.",
+    )
+    status: Literal[
+        "success",
+        "excluded_domain",
+        "http_error",
+        "network_error",
+        "unsupported_method",
+    ] = Field(
+        ...,
+        description="The status of the operation. 'success' means a 2xx status code; 'http_error' means a 4xx/5xx code; 'network_error' means connection failed.",
+    )
+    error_message: Optional[str] = Field(
+        None, description="An error message if the status is not 'success'."
+    )
+
+
+@Tool
+def http_get_tool(
+    ctx: RunContext,
+    input: GetURLContentInput,
+) -> URLContent:
+    """
+    Retrieve content of a given URL via a simple HTTP GET request.
+    This tool only supports HTTP and HTTPS GET methods. It does not follow
+    redirects and has a fixed, short timeout.
+    """
+    url = input.url
+    logger.info(f"http_get_tool(url='{url}')")
+
+    try:
+        if not url.startswith("http://") and not url.startswith("https://"):
+            return URLContent(
+                requested_url=url,
+                status="unsupported_method",
+                error_message="URL must start with 'http://' or 'https://'.",
+            )
+
+        parsed_url = urlparse(url)
+        hostname = parsed_url.netloc.lower()
+        for allowed_host in allow_list:
+            if allowed_host not in hostname:
+                logger.warning(
+                    f"http_get tool is not permitted to access domain: '{hostname}'."
+                )
+                return URLContent(
+                    requested_url=url,
+                    status="excluded_domain",
+                    error_message=f"Domain '{hostname}' is not permitted.",
+                )
+
+        response = requests.get(
```

Comment:
Should we limit the size of received/processed data somewhere?

Reply:
not just yet, I have a general task (across all tools) related to context token size (which relates to this) and usage and retries ... which should be done as part of that PR.